util/diff_utils.go

```diff
@@ -118,6 +123,36 @@ func DiffDirectory(d1, d2 pkgutil.Directory) (DirDiff, bool) {
 	return DirDiff{addedEntries, deletedEntries, modifiedEntries}, same
 }
 
+//DiffFile diffs within a file
+func DiffFile(path1, path2, image1, image2 string) (FileNameDiff, error) {
```

Comment:
This doesn't satisfy any interface right?  In that case, you should probably pass the filename parameter into here too and set it on the FileNameDiff